run-custom-mao.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash -xe
+
+export SCRIPTS_DIR=$(pwd)
+export CUSTOM_MAO_BUILD=quay.io/sdasu/machine-api-operator:sdasu-fix
+
+kubectl get secrets --namespace openshift-machine-api | grep mariadb
+
+# Stop CVO, MAO, machine-api-controller and Metal3 deployments
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 cluster-version-operator --namespace openshift-cluster-version
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment -n openshift-machine-api --replicas=0 machine-api-operator
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 machine-api-controllers --namespace openshift-machine-api
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 metal3 --namespace openshift-machine-api
+
+REPLACEMENT_TEXT="image: ${CUSTOM_MAO_BUILD}"
+echo $REPLACEMENT_TEXT
+
+oc get deployments -n openshift-machine-api machine-api-operator -o yaml > mao.yaml
+sed -i -e "s!image: .*!${REPLACEMENT_TEXT}!g" mao.yaml
+oc apply -f mao.yaml
+
+# Give the machine-api-operator rbac access to create and read secrets
+oc apply -f machine-api-operator-roles.yaml
+
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment -n openshift-machine-api --replicas=1 machine-api-operator
```

Comment:
I don't think you need these steps. Replacing the machine-api-operator deployment definition one line 19 should have changed the replica count back to 1. I would expect that to cause the new MAO to run, and for it to update the deployments for the machine-api-controllers and metal3. Does it not do that?

Reply:
Yes, once we scale the machine-api-operator back up, the reconcile process does deploy machine-api-controllers and metal3. Specifically scaling it back up this way starts their deployment right away. Will remove these 2 lines.